File: Java/프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        for (int i = s.length(); i > 0; i--) {
            for (int j = 0; j + i <= s.length(); j++) {
                if (isPalindrome(j, j + i - 1, s)) {
                    return i;
                }
            }
        }
        return answer;
    }

    boolean isPalindrome(int start, int end, String s) {
        while (start <= end) {
            if (s.charAt(start++) != s.charAt(end--)) {
                return false;
            }
        }
        return true;
    }
}

File: Python/프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.py
def solution(s):
    def is_palindrome(sub):
        reverse = sub[::-1]
        return sub == reverse

    answer = 0

    for i in range(len(s), 0, -1):
        j = 0
        while j + i <= len(s):
            sub = s[j : j + i]
            if is_palindrome(sub):
                return len(sub)
            j += 1

    return answer